### PR TITLE
feat(VNavigationDrawer): update to md2

### LINKED ISSUE #2606: [Bug Report] Navigation drawer has incorrect icon alignment when custom mini-variant-width is set
### Versions and Environment
**Vuetify:** 0.17.2
**Vue:** 2.5.7
**Browsers:** Chrome 62.0.3202.94
**OS:** Windows 7

### Steps to reproduce
Create a navigation drawer with some list tiles and set the `mini-variant-width` to a non-default value (e.g. 100). Now activate mini-variant.

See provided codepen for issue demonstration and fix.

### Expected Behavior
Tile icons should be horizontally centered regardless of the width of the mini sidebar.

### Actual Behavior
The icons are off-center because it assumes the default `mini-variant-width` of 80px to calculate the icon alignment. Also, if you do not wrap the `v-list-tile-title` in a `v-list-tile-content` parts of the title will still show in mini mode because the opacity override only applies to `.list__tile__content`. Same with the right hand icon (the up/down arrow) in case of using a `list-tile-group`.

### Reproduction Link
[https://codepen.io/anon/pen/pdLXWj](https://codepen.io/anon/pen/pdLXWj)

### Other comments
I am proposing a fix in my codepen which will make it work with any sidebar width. Also, with this solution the current CSS overrides to hide title, etc. is not technically needed because the icon/avatar will always fill the full width of the sidebar.

### LINKED ISSUE #2830: [Bug Report] Grouped list looks wrong on mini variant if opened before
### Versions and Environment
**Vuetify:** 1.0.0-alpha.3
**Vue:** 2.5.11
**Browsers:** Chrome 63.0.3239.108
**OS:** Linux x86_64

### Steps to reproduce
1. Open to extended view in navigation by clicking on the arrow icon
2. Open the grouped list "Content Management"
3. Close the mini variant by clicking on the arrow icon

=> 4. Now you will see the bug: There is an empty space after the "content management" icon in the mini variant.. this looks wrong to me.

### Expected Behavior
In mini variant the items in the group list should not be shown as empty spaces.

### Actual Behavior
In the mini variant the items in the group list may be shown as empty spaces if opened before enabling the mini variant mode.

### Reproduction Link
[https://codepen.io/anon/pen/Rxarqw](https://codepen.io/anon/pen/Rxarqw)

bildschirmfoto von 2017-12-19 10-02-16

### LINKED ISSUE #3146: [Bug Report] Unable to scroll mini-variant drawer
### Versions and Environment
**Vuetify:** 1.0.0-beta.5
**Vue:** 2.5.13
**Browsers:** Chrome 63.0.3239.132
**OS:** Mac OS 10.13.3

### Steps to reproduce
Create a mini-variant toolbar with menu items that extend past the page and try to scroll

### Expected Behavior
The menu be scrollable

### Actual Behavior
The menu is not scrollable

### Reproduction Link
[https://codepen.io/mattchewone/pen/jZEvZE](https://codepen.io/mattchewone/pen/jZEvZE)

### Other comments
https://github.com/vuetifyjs/vuetify/blob/2ac4927d4b2e11f654e067c5a743a14589a4d011/src/stylus/components/_navigation-drawer.styl#L65-L66

### LINKED ISSUE #3498: [Bug Report] Border on navigation-drawer missing with scrollbar
### Versions and Environment
**Vuetify:** 1.0.5
**Vue:** 2.5.13
**Browsers:** Chrome 64.0.3282.186
**OS:** Mac OS 10.13.3

### Steps to reproduce
Or Go to https://vuetifyjs.com/en/components/alerts and scroll up in the drawer

### Expected Behavior
Border should be displayed all the time

### Actual Behavior
Border ends after content which was visible on load

### Reproduction Link
[https://jsfiddle.net/kLtkzbxk/51/](https://jsfiddle.net/kLtkzbxk/51/)

### LINKED ISSUE #3536: [Feature Request] Allow to use v-spacer inside v-navigation-drawer to make the elements "bottom-aligned"
### New Functionality
It would allow to move list elements (inside of a drawer) to the bottom & to scroll the whole list on y-overflow:

- **See GIF** (it's 60mb, let it load): https://imgur.com/a/NDbN3
- **Codepen:** https://codepen.io/anon/pen/zWGwOe?editors=1000

### Improvements
More use cases (for example: Windows 10 UWP style drawer). 
Right now, `v-spacer` doesn't work inside of `v-navigation-drawer`

### Bugs or Edge Cases it Helps Avoid
none

### LINKED ISSUE #3919: [Feature Request] Support "vw" and "vh" for navigation-drawer
# Problem to solve
The "length" parameter (https://vuetifyjs.com/en/components/navigation-drawers) does not support "vw" and "vh".

# Proposed solution
Being able to support % and vw/vh to give alternatives to pixels parameters

Allow the parameter to be used as a string. 

Exemple : 
```
<v-navigation-drawer
    fixed
    clipped
    v-model="drawer"
    right
    app
    :disable-resize-watcher="true"
    width="20vw"
  >
  </v-navigation-drawer>
```

Assuming that we have "vw", "vh" or "%", use them directly as string into style balise instead of number parameter for width in pixels

### LINKED ISSUE #4233: [Bug Report] Navigation Drawer - Stateless prop doesn't disable touch gestures
### Versions and Environment
**Vuetify:** 1.0.19
**Vue:** 2.5.16
**Browsers:** Opera 53.0.2907.68, Google Chrome
**OS:** Windows 10

### Steps to reproduce
1. Open the codepen provided on mobile, or using a browser with gestures enabled
2. Click on the menu icon to activate the parent drawer
3. Use a `right-to-left` swiping gesture to close the parent drawer
4. Re-open the parent drawer by clicking on the menu icon (nested drawer is now hidden)
5. Use a `left-to-right` swiping gesture to show the nested drawer

### Expected Behavior
A nested `<v-navigation-drawer>` with `stateless` prop should not hide or show based on gestures.

### Actual Behavior
A nested `<v-navigation-drawer>` with `stateless` prop applied hides or shows when a swiping gesture is used

### Reproduction Link
[https://codepen.io/ocolas/pen/ZRBEXB](https://codepen.io/ocolas/pen/ZRBEXB)

### Other comments
Combined drawers, one navigation-drawer inside another. 
The nested navigation-drawer has props `mini-variant, stateless, value="true"`

### LINKED ISSUE #6283: [Feature Request] Theme-able color for navigation drawer
### Problem to solve
Current color for navigation drawer is the `$material.cards`. I would like to see possibility to override theme (dark and light) so that navigation drawer could be the same color as application background, and cards would still be in different color.

### Proposed solution
Add a color for navigation drawer in theme, and use that color in `_navigation-drawer.styl`. Default color could still be the same as `$material.cards`.

### LINKED ISSUE #6652: [Bug Report] Inconsistent collapse animation in navigation drawer with group lists
### Versions and Environment
**Vuetify:** 1.5.4
**Vue:** 2.5.16
**Browsers:** Google Chrome, Mozilla Firefox
**OS:** Linux

### Steps to reproduce
1. Have a navigation drawer with grouped and non-grouped items, all of them with icons.
2. Activate mini-variant

### Expected Behavior
The group activator tiles do the same animation as the simple tiles (text disappearing from the right with static icon)

### Actual Behavior
The group activator icon jumps to the right and then animates to the left.

### Reproduction Link
<a href="https://codepen.io/anon/pen/aMdPJa" target="_blank">https://codepen.io/anon/pen/aMdPJa</a>

### Other comments
This is probably because we're forced to use `:prepend-icon` on group headers, because if we use a regular icon on the activator template it disappears on collapse.

### PR DESCRIPTION
**Pull request by johnleider**
## Description
Update to match md2 spec.

fixes #2606 
fixes #2830
fixes #3146
fixes #3498
resolves #3536 
resolves #3919
resolves #4233
resolves #6283 
resolves #6652

## Motivation and Context
Upgrade framework to md2

## How Has This Been Tested?
jest

## Markup:

<details>

```vue
<template>
  <div class="ma-5 pa-5">
    <v-navigation-drawer
      v-model="drawer"
      app
    >
      <template v-slot:prepend>
        <v-toolbar flat>
          <v-list-item two-line>
            <v-list-item-content>
              <v-list-item-title class="title">Title</v-list-item-title>
              <v-list-item-subtitle>subtext</v-list-item-subtitle>
            </v-list-item-content>
          </v-list-item>
        </v-toolbar>
      </template>
      <v-list
        nav
        dense
      >
        <v-list-item-group>
          <v-list-item>
            <v-list-item-icon>
              <v-icon>mdi-inbox</v-icon>
            </v-list-item-icon>
            <v-list-item-content>
              <v-list-item-title>Inbox</v-list-item-title>
            </v-list-item-content>
          </v-list-item>
          <v-list-item>
            <v-list-item-icon>
              <v-icon>mdi-star</v-icon>
            </v-list-item-icon>
            <v-list-item-content>
              <v-list-item-title>Star</v-list-item-title>
            </v-list-item-content>
          </v-list-item>
          <v-list-item>
            <v-list-item-icon>
              <v-icon>mdi-send</v-icon>
            </v-list-item-icon>
            <v-list-item-content>
              <v-list-item-title>Sent Mail</v-list-item-title>
            </v-list-item-content>
          </v-list-item>
          <v-list-item>
            <v-list-item-icon>
              <v-icon>mdi-email-open</v-icon>
            </v-list-item-icon>
            <v-list-item-content>
              <v-list-item-title>Drafts</v-list-item-title>
            </v-list-item-content>
          </v-list-item>
        </v-list-item-group>
      </v-list>

      <template v-slot:append>
        <div class="pa-2">
          <v-btn
            block
            color="primary"
          >Buy Pro</v-btn>
        </div>
      </template>
    </v-navigation-drawer>

    <v-app-bar app>
      <v-app-bar-nav-icon></v-app-bar-nav-icon>
      <v-toolbar-title>Travel</v-toolbar-title>
    </v-app-bar>

    <v-btn @click.stop="drawer = !drawer">Click me</v-btn>

    <v-footer
      app
      inset
    >
      I'm content
    </v-footer>
  </div>
</template>

<script>
  export default {
    data: () => ({
      drawer: true
    })
  }
</script>

```
</details>

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Improvement/refactoring (non-breaking change that doesn't add any feature but make things better)

## Checklist:

- [x] The PR title is no longer than 64 characters.
- [x] The PR is submitted to the correct branch (`master` for bug fixes and documentation updates, `dev` for new features and breaking changes).
- [x] My code follows the code style of this project.
- [ ] I've added relevant changes to the documentation (applies to new features and breaking changes in core library)

### CODE REVIEW

File: packages/vuetify/src/components/VNavigationDrawer/VNavigationDrawer.sass [hunk 1]
```diff
@@ -11,67 +11,110 @@
   .v-divider
     border-color: map-get($material, 'dividers')
 
+// Block
 .v-navigation-drawer
-  +bootable()
-
-  display: block
+  -webkit-overflow-scrolling: touch
```
Comment (sh7dm) [inline packages/vuetify/src/components/VNavigationDrawer/VNavigationDrawer.sass]: We use autoprefixer, looks like we shouldn't use prefixes here.

File: packages/vuetify/src/components/VNavigationDrawer/VNavigationDrawer.sass [hunk 2]
```diff
@@ -11,67 +11,110 @@
...
   left: 0
   max-width: 100%
-  overflow-y: auto
-  overflow-x: hidden
+  overflow: hidden
   pointer-events: auto
   top: 0
-  will-change: transform
-  -webkit-overflow-scrolling: touch
   transition-duration: 0.2s
```
Comment (sh7dm) [inline packages/vuetify/src/components/VNavigationDrawer/VNavigationDrawer.sass]: Variables?

File: packages/vuetify/src/components/VNavigationDrawer/VNavigationDrawer.sass [hunk 3]
```diff
@@ -11,67 +11,110 @@
...
+
+    > *:first-child
+      margin-left: 0
+      margin-right: 0
+
+    > *:not(:first-child)
+      display: none
+
+  .v-list-group--no-action
+    > .v-list-group__items
```
Comment (sh7dm) [inline packages/vuetify/src/components/VNavigationDrawer/VNavigationDrawer.sass]: No need for nested selectors

File: packages/vuetify/src/components/VNavigationDrawer/VNavigationDrawer.ts [hunk 4]
```diff
@@ -14,25 +18,33 @@ import Resize from '../../directives/resize'
...
+import { convertToUnit, getSlot } from '../../util/helpers'
 import mixins from '../../util/mixins'
 
 // TYpes
-import { VNode } from 'vue/types/vnode'
+import { VNode, VNodeDirective } from 'vue'
 import { PropValidator } from 'vue/types/options'
 
-export default mixins(
+const baseMixins = mixins(
```
Comment (sh7dm) [inline packages/vuetify/src/components/VNavigationDrawer/VNavigationDrawer.ts]: Why is it required?

File: packages/vuetify/src/components/VNavigationDrawer/VNavigationDrawer.sass (hunk 1 above)
Comment (johnleider) [reply packages/vuetify/src/components/VNavigationDrawer/VNavigationDrawer.sass]: This is only supported as the above in browsers that it is supported in.

File: packages/vuetify/src/components/VNavigationDrawer/VNavigationDrawer.sass (hunk 2 above)
Comment (johnleider) [reply packages/vuetify/src/components/VNavigationDrawer/VNavigationDrawer.sass]: There is nothing worth adding to a variable here as the user can either change it locally through props or isn't something I feel would be used very much.

File: packages/vuetify/src/components/VNavigationDrawer/VNavigationDrawer.ts (hunk 4 above)
Comment (johnleider) [reply packages/vuetify/src/components/VNavigationDrawer/VNavigationDrawer.ts]: This is the similar style we use in the rest of the framework.

**Pull request merged**